ArduPlane/commands_logic.cpp

```diff
@@ -1071,7 +1060,19 @@ bool Plane::verify_landing_vtol_approach(const AP_Mission::Mission_Command &cmd)
                 // check if we should move on to the next waypoint
                 Location breakout_loc = cmd.content.location;
                 breakout_loc.offset_bearing(vtol_approach_s.approach_direction_deg + 180, quadplane.stopping_distance());
-                if(current_loc.past_interval_finish_line(start, breakout_loc)) {
+
+                const bool past_finish_line = current_loc.past_interval_finish_line(start, breakout_loc);
+                const bool half_radius = current_loc.get_distance(cmd.content.location) < 0.5 * abs_radius;
+                bool lined_up = true;
+                Vector3f vel_NED;
+                if (ahrs.get_velocity_NED(vel_NED)) {
+                    const float ground_course_deg = wrap_180(degrees(vel_NED.xy().angle()));
+                    const float approach_deg = wrap_180(vtol_approach_s.approach_direction_deg);
+                    const float angle_err = fabsf(wrap_180(ground_course_deg - approach_deg));
+                    lined_up = (angle_err < 30);
```

Comment:
This assumes that you approach at the calculated approach direction. In reality we turn towards target at that angle. Depending on your turn radius you might not be inside the 30 deg tollerance.

How about comparing to the velocity vector to the direction of the target? Something like:
```
Vector2f target_vec = current_loc.get_distance_NE(cmd.content.location);
const float angle_err = vel_NED.xy().angle(target_vec);
lined_up = (angle_err < 30);
```
 
If you keep you original method you only need the last `wrap_180`

Reply:
good, idea, changed